internal/service/vpc/resource_aws_vpc_peering_connection.go

```diff
@@ -0,0 +1,337 @@
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	pc, err := client.VPCPeeringConnections.GetVPCPeering(
+		projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+	}
+
+	return copyAWSVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(d, pc, projectName, vpcID)
+}
+
+func resourceAWSVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+
+	err := client.VPCPeeringConnections.DeleteVPCPeering(
+		projectName,
+		vpcID,
+		peerCloudAccount,
+		peerVPC,
+		peerRegion,
+	)
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.Errorf("Error deleting VPC peering connection: %s", err)
+	}
+
+	stateChangeConf := &resource.StateChangeConf{
+		Pending: []string{
+			"ACTIVE",
+			"APPROVED",
+			"APPROVED_PEER_REQUESTED",
+			"DELETING",
+			"INVALID_SPECIFICATION",
+			"PENDING_PEER",
+			"REJECTED_BY_PEER",
+			"DELETED_BY_PEER",
+		},
+		Target: []string{
+			"DELETED",
+		},
+		Refresh: func() (interface{}, string, error) {
+			pc, err := client.VPCPeeringConnections.GetVPCPeering(
+				projectName,
+				vpcID,
+				peerCloudAccount,
+				peerVPC,
+				peerRegion,
+			)
+			if err != nil {
+				return nil, "", err
+			}
+			return pc, pc.State, nil
+		},
+		Delay:      10 * time.Second,
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		MinTimeout: 2 * time.Second,
+	}
+	if _, err := stateChangeConf.WaitForStateContext(ctx); err != nil && !aiven.IsNotFound(err) {
+		return diag.Errorf("Error waiting for Aiven AWS VPC Peering Connection to be DELETED: %s", err)
+	}
+	return nil
+}
+
+func resourceAWSVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	if len(strings.Split(d.Id(), "/")) != 3 {
+		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>", d.Id())
+	}
+
+	dig := resourceAWSVPCPeeringConnectionRead(ctx, d, m)
+	if dig.HasError() {
+		return nil, errors.New("cannot get aws vpc peering connection")
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func copyAWSVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(
+	d *schema.ResourceData,
+	peeringConnection *aiven.VPCPeeringConnection,
+	project string,
+	vpcID string,
+) diag.Diagnostics {
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	if err := d.Set("vpc_id", schemautil.BuildResourceID(project, vpcID)); err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  fmt.Sprintf("Unable to set vpc_id field: %s", err),
+			Detail:   fmt.Sprintf("Unable to set vpc_id field for AWS VPC peering connection: %s", err),
+		})
```

Comment:
Probably at least `diag.Diagnostic` can be created inside a function with `field_name` and `err` arguments.

Reply:
indeed, there is no value in what we are doing since it can be generated :+1: